### PR TITLE
[No. Series Copilot] Bundled modifications to the No. Series Copilot

### PR DESCRIPTION
**Pull request by JesperSchulz**
#### Summary
This PR addesses the following issues:
- We should not return free-text from the model. Free-text replies have hence been replaced by canned error messages - which also render better in the prompt dialog now.
- The feature registration may not include the work Copilot. The capability is not a copilot, it a capability of the Copilot.
- Add simple AI test automation added.

#### Work Item(s) 
Fixes [AB#544083](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/544083), [AB#544055](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/544055)

### CODE REVIEW

File: src/Business Foundation/Test Library/NoSeriesCopilot/app.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (JesperSchulz) [inline src/Business Foundation/Test Library/NoSeriesCopilot/app.json:1]: Todo: The app.jsons need some love.

File: src/Business Foundation/App/NoSeriesCopilot/src/Copilot/NoSeriesCopilotImpl.Codeunit.al [hunk 2]
```diff
@@ -61,7 +61,8 @@ codeunit 324 "No. Series Copilot Impl."
                 SaveGenerationHistory(NoSeriesGeneration, InputText);
                 CreateNoSeries(NoSeriesGeneration, GeneratedNoSeries, Completion);
             end else
-                ResponseText := Completion;
+                if Completion <> '' then
```
Comment (darjoo) [inline src/Business Foundation/App/NoSeriesCopilot/src/Copilot/NoSeriesCopilotImpl.Codeunit.al]: What happens if `Completion` is empty? I'd expect this else check to just error irregardless of what Completion has.

File: src/Business Foundation/App/NoSeriesCopilot/src/Copilot/NoSeriesCopilotImpl.Codeunit.al [hunk 3]
```diff
@@ -171,7 +172,7 @@ codeunit 324 "No. Series Copilot Impl."
         if not AzureOpenAI.IsEnabled(Enum::"Copilot Capability"::"No. Series Copilot") then
             exit;
 
-        AzureOpenAI.SetAuthorization(Enum::"AOAI Model Type"::"Chat Completions", AOAIDeployments.GetGPT4oMiniLatest());
+        AzureOpenAI.SetAuthorization(Enum::"AOAI Model Type"::"Chat Completions", AOAIDeployments.GetGPT35TurboPreview());
```
Comment (darjoo) [inline src/Business Foundation/App/NoSeriesCopilot/src/Copilot/NoSeriesCopilotImpl.Codeunit.al]: Pull latest, my changes for using GPT4oMiniLatest is in.

File: src/Business Foundation/Test Library/NoSeriesCopilot/src/LibraryNoSeriesCopilot.Codeunit.al [hunk 4]
```diff
@@ -0,0 +1,13 @@
+namespace Microsoft.Foundation.NoSeries.Test;
```
Comment (darjoo) [inline src/Business Foundation/Test Library/NoSeriesCopilot/src/LibraryNoSeriesCopilot.Codeunit.al:6]: Missing copyright

File: src/Business Foundation/Test/NoSeriesCopilot/src/NoSeriesCopilotAccuTests.Codeunit.al [hunk 5]
```diff
@@ -0,0 +1,33 @@
+namespace Microsoft.Foundation.NoSeries.Test;
```
Comment (darjoo) [inline src/Business Foundation/Test/NoSeriesCopilot/src/NoSeriesCopilotAccuTests.Codeunit.al:6]: Missing copyright

**Review by darjoo — DISMISSED** [review]

**Review by darjoo — APPROVED** [review]

**Review by stkillen — APPROVED** [review]

**Pull request merged**